lib/doorkeeper/request/password.rb

```diff
@@ -13,16 +13,6 @@ def request
           parameters
         )
       end
-
-      private
-
-      def client
-        if credentials
-          server.client
-        elsif parameters[:client_id]
-          server.client_via_uid
```

Comment:
Is it that we removed client_via_uid and introduced only client?

Reply:
I think so, because this method was originally doing was allowing `client` to be set if:

1. Valid credentials aka good `client_id` and `client_secret`
2. Only good `client_id` set 

Because `credentials` now returns true if good `client_id` and `client_secret` *or* only good `client_id` set this private method is redundant. We can trust the `server.client` to be set when either (1) or (2) is satified.